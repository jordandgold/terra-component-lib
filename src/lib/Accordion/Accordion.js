import React, { Component } from "react";
import Icon from "../Icon/Icon";
import "./Accordion.scss";
import PropTypes from "prop-types";

class Accordion extends Component {
  constructor(props) {
    super(props);

    this.state = {
      active: undefined
    };
  }

  checkActive = () => {
    if (this.props.defaultActive === 0 || this.props.defaultActive) {
      this.setState({
        active: this.props.defaultActive
      });
    }
  };

  accordionToggle = fold => {
    if (this.state.active === fold) {
      this.setState({
        active: undefined
      });
    } else {
      this.setState({ active: fold });
    }
  };

  generateFolds = () => {
    return this.props.children.map((child, index) => {
      let activeClass = index == this.state.active ? "is-expanded" : "";
      return (
        <div
          key={`accordion-fold-${index}`}
          className={`ter-accordion__fold ${activeClass}`}
        >
          <button
            className="ter-accordion__trigger"
            onClick={() => this.accordionToggle(index)}
          >
            {child.props.title}
            <Icon
              type="open-caret-right-dark-16px"
              className="ter-accordion__trigger-caret"
              size="16px"
            />
          </button>
          {child}
        </div>
      );
    });
  };

  componentDidMount() {
    this.checkActive();
  }

  render() {
    return <div className="ter-accordion">{this.generateFolds()}</div>;
  }
}

export default Accordion;

export class AccordionFold extends Component {
  render() {
    return (
      <React.Fragment>
        <div className="ter-accordion__content">{this.props.children}</div>
      </React.Fragment>
    );
  }
}

Accordion.propTypes = {
  defaultActive: PropTypes.number,
  children: PropTypes.node.isRequired
};
