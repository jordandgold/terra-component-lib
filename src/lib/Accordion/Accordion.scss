@import "../global-styles/terra.scss";

.ter-accordion {
  margin-bottom: spacing("sm");
  border-bottom: $accordion-border-width solid $accordion-border-color;
  &__fold {
    font-size: $accordion-font-size;
    margin-bottom: spacing("sm");
    border-color: $accordion-border-color;
    border-width: $accordion-border-width;
    border-style: solid;
    border-left-width: 0px;
    border-right-width: 0px;
    transition: background 225ms ease-in-out;
    margin-bottom: 0;
    border-bottom: 0px;
    &:first-child {
      border-bottom: 0px;
    }
    &:last-child {
      border-bottom: 0px;
    }
  }
  &__trigger {
    @include button-reset();
    font-size: $accordion-font-size;
    padding: spacing("xs") spacing("lg") spacing("xs") spacing("sm");
    position: relative;
    display: block;
    text-decoration: none;
    font-weight: font-weight("bold");
    width: 100%;
    text-align: left;
    line-height: line-height("loose");
    &-caret {
      position: absolute;
      color: $accordion-arrow-color;
      top: 50%;
      right: spacing("sm");
      transform: rotate(0deg) translateY(-50%);
      transition: transform 200ms ease-in-out;
      font-size: 16px;
      width: 16px;
      height: 16px;
    }
    &:hover {
      cursor: pointer;
      text-decoration: none;
    }
  }
  &__content {
    font-size: $accordion-font-size;
    display: none;
    padding: spacing("xs") spacing("sm");
    p {
      font-size: $accordion-font-size;
    }
  }
  &__fold {
    &.is-expanded {
      .ter-accordion__content {
        display: block;
      }
      .ter-accordion__trigger {
        text-decoration: none;
        &-caret {
          transform: rotate(90deg) translateX(-50%);
        }
      }
    }
  }
}
