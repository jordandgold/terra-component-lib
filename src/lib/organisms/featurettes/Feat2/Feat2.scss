@import "../../../global-styles/terra.scss";

.ter-feat-two {
  padding: spacing("lg");

  &__image--mobile {
    margin: auto;
    display: block;
    width: 100%;
    height: auto;

    @media (min-width: $breakpoint-sm) {
      display: none;
    }
  }

  &__title {
    text-align: center;
    font-size: font-size("lg");
    max-width: 800px;
    margin: spacing("md") auto;
  }

  &__text {
    max-width: 800px;
    margin: spacing("md") auto;
  }

  &__accordion-image-container {
    @media (min-width: $breakpoint-sm) {
      display: flex;
      justify-content: space-around;
      padding: spacing("md");
    }
  }

  &__accordion-wrapper {
    @media (min-width: $breakpoint-sm) {
      padding: 0 spacing("md");
      width: 100%;
    }

    .ter-accordion {
      align-self: start;
    }
  }

  &__image {
    margin: auto;
    display: block;
    width: 100%;
  }

  &__image-wrapper {
    display: none;

    width: 100%;
    height: auto;
    margin: auto;
    padding-bottom: spacing("lg");

    @media (min-width: $breakpoint-sm) {
      padding: 0 spacing("md");
      display: block;
    }

    @media (min-width: $breakpoint-sm) {
      padding-bottom: 0;
    }
  }
}
