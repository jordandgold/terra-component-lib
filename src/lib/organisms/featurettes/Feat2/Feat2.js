import React from "react";
import PropTypes from "prop-types";
import Accordion from "../../../Accordion/Accordion";
import "./Feat2.scss";

const Feat2 = ({ title, text, children, image }) => {
  return (
    <section className="ter-feat-two">
      <img
        className="ter-feat-two__image--mobile"
        src={image.url}
        alt={image.altText}
      />
      <h2 className="ter-feat-two__title">{title}</h2>
      <p className="ter-feat-two__text">{text}</p>
      <div className="ter-feat-two__accordion-image-container">
        <div className="ter-feat-two__image-wrapper">
          <img
            className="ter-feat-two__image-desktop ter-feat-two__image"
            src={image.url}
            alt={image.altText}
          />
        </div>
        <div className="ter-feat-two__accordion-wrapper">
          <Accordion defaultActive={0} children={children} />
        </div>
      </div>
    </section>
  );
};

export default Feat2;

Feat2.propTypes = {
  title: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  image: PropTypes.shape({
    url: PropTypes.string.isRequired,
    altText: PropTypes.string.isRequired
  }).isRequired
};
