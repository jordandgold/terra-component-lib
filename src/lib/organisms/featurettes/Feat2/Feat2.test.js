import React from "react";
import { shallow } from "enzyme";
import Feat2 from "./Feat2";
import { AccordionFold } from "../../../Accordion/Accordion";

describe("Feat2", () => {
  let wrapper;

  beforeEach(() => {
    const mockImage = { url: "./1-to-1.png", altText: "Placeholder image" };
    const mockText =
      "Eu elit non Lorem deserunt sint aute aliquip esse non sint tempor deserunt voluptate reprehenderit. Duis duis aute sint tempor proident officia enim aliqua enim sit exercitation. Ex culpa dolor ex reprehenderit adipisicing. Magna dolore occaecat nisi voluptate sunt qui nulla cupidatat minim sit non nisi exercitation occaecat.";

    wrapper = shallow(
      <Feat2
        image={mockImage}
        title="Good design is a thing, apparently."
        text={mockText}
      >
        <AccordionFold title="Test one">
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer
            lobortis augue ligula, eget gravida tellus lacinia id. Sed ultricies
            mi malesuada tincidunt dapibus. Donec porta ligula sagittis elit
            sollicitudin.
          </p>
        </AccordionFold>
        <AccordionFold title="Test two">
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer
            lobortis augue ligula, eget gravida tellus lacinia id. Sed ultricies
            mi malesuada tincidunt dapibus. Donec porta ligula sagittis elit
            sollicitudin.
          </p>
        </AccordionFold>
        <AccordionFold title="Test three">
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer
            lobortis augue ligula, eget gravida tellus lacinia id. Sed ultricies
            mi malesuada tincidunt dapibus. Donec porta ligula sagittis elit
            sollicitudin.
          </p>
        </AccordionFold>
      </Feat2>
    );
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
