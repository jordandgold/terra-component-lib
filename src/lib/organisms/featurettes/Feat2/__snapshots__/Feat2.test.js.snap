// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feat2 should match the snapshot 1`] = `
<section
  className="ter-feat-two"
>
  <img
    alt="Placeholder image"
    className="ter-feat-two__image--mobile"
    src="./1-to-1.png"
  />
  <h2
    className="ter-feat-two__title"
  >
    Good design is a thing, apparently.
  </h2>
  <p
    className="ter-feat-two__text"
  >
    Eu elit non Lorem deserunt sint aute aliquip esse non sint tempor deserunt voluptate reprehenderit. Duis duis aute sint tempor proident officia enim aliqua enim sit exercitation. Ex culpa dolor ex reprehenderit adipisicing. Magna dolore occaecat nisi voluptate sunt qui nulla cupidatat minim sit non nisi exercitation occaecat.
  </p>
  <div
    className="ter-feat-two__accordion-image-container"
  >
    <div
      className="ter-feat-two__image-wrapper"
    >
      <img
        alt="Placeholder image"
        className="ter-feat-two__image-desktop ter-feat-two__image"
        src="./1-to-1.png"
      />
    </div>
    <div
      className="ter-feat-two__accordion-wrapper"
    >
      <Accordion
        defaultActive={0}
      >
        <AccordionFold
          title="Test one"
        >
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer lobortis augue ligula, eget gravida tellus lacinia id. Sed ultricies mi malesuada tincidunt dapibus. Donec porta ligula sagittis elit sollicitudin.
          </p>
        </AccordionFold>
        <AccordionFold
          title="Test two"
        >
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer lobortis augue ligula, eget gravida tellus lacinia id. Sed ultricies mi malesuada tincidunt dapibus. Donec porta ligula sagittis elit sollicitudin.
          </p>
        </AccordionFold>
        <AccordionFold
          title="Test three"
        >
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer lobortis augue ligula, eget gravida tellus lacinia id. Sed ultricies mi malesuada tincidunt dapibus. Donec porta ligula sagittis elit sollicitudin.
          </p>
        </AccordionFold>
      </Accordion>
    </div>
  </div>
</section>
`;
